scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,390 @@
+			os.environ["XTENSA_SYSTEM"] = XTENSA_SYSTEM
+			print(f"XTENSA_SYSTEM={XTENSA_SYSTEM}")
+
+		platform_build_dir_name = f"build-{platform}"
+		PLAT_CONFIG = platform_dict["PLAT_CONFIG"]
+		build_cmd = ["west"]
+		if args.verbose > 0:
+			build_cmd.append("-v")
+		build_cmd += ["build", "--build-dir", platform_build_dir_name]
+		source_dir = pathlib.Path(west_top, "zephyr", "samples", "subsys", "audio", "sof")
+		build_cmd += ["--board", PLAT_CONFIG, str(source_dir)]
+		if args.cmake_args:
+			build_cmd += ["--", args.cmake_args]
+		# Build
+		execute_command(build_cmd, check=True, cwd=west_top)
+		smex_executable = pathlib.Path(west_top, platform_build_dir_name, "zephyr", "smex_ep",
+			"build", "smex")
+		fw_ldc_file = pathlib.Path(STAGING_DIR, f"sof-{platform}.ldc")
+		input_elf_file = pathlib.Path(west_top, platform_build_dir_name, "zephyr", "zephyr.elf")
+		# Extract metadata
+		execute_command([str(smex_executable), "-l", str(fw_ldc_file), str(input_elf_file)],
+			check=True)
+		# Update SOF submodules
+		git_submodules_update()
+		# CMake - configure rimage module
+		rimage_dir_name="build-rimage"
+		sof_mirror_dir = pathlib.Path("modules", "audio", "sof")
+		rimage_source_dir = pathlib.Path(sof_mirror_dir, "rimage")
+		execute_command(["cmake", "-B", rimage_dir_name, "-S", str(rimage_source_dir)], check=True,
+			cwd=west_top)
+		# CMake build rimage module
+		execute_command(["cmake", "--build", rimage_dir_name, "-j", args.jobs], check=True,
+			cwd=west_top)
+		# Sign firmware
+		rimage_executable = shutil.which("rimage", path=pathlib.Path(west_top, rimage_dir_name))
+		rimage_config = pathlib.Path(sof_mirror_dir, "rimage", "config")
+		sign_cmd = ["west", "sign", "--build-dir", platform_build_dir_name, "--tool", "rimage"]
+		sign_cmd += ["--tool-path", rimage_executable]
+		signing_key = ""
+		if args.key:
+			signing_key = args.key
+		elif "RIMAGE_KEY" in platform_dict:
+			signing_key = platform_dict["RIMAGE_KEY"]
+		else:
+			signing_key = default_rimage_key
+		sign_cmd += ["--tool-data", str(rimage_config), "--", "-k", signing_key]
+		execute_command(sign_cmd, check=True, cwd=west_top)
+		# Install by copy
+		fw_file_to_copy = pathlib.Path(west_top, platform_build_dir_name, "zephyr", "zephyr.ri")
+		fw_file_installed = pathlib.Path(STAGING_DIR, "community", f"sof-{platform}.ri")
+		shutil.copy2(str(fw_file_to_copy), str(fw_file_installed))
+
+	# Install sof-logger
+	sof_logger_executable_to_copy = pathlib.Path(west_top, platform_build_dir_name, "zephyr",
+		"sof-logger_ep", "build", "logger", "sof-logger")
+	sof_logger_installed = pathlib.Path(STAGING_DIR, "..", "tools", "sof-logger").resolve()
+	shutil.copy2(str(sof_logger_executable_to_copy), str(sof_logger_installed))
+
+def run_clone_mode():
+	if find_west_workspace():
+		raise RuntimeError("Zephyr found already! Not downloading it again")
+	create_zephyr_directory()
+	west_init_update()
+	git_submodules_update()
+	build_platforms()
```

Comment:
`build_platforms()` belongs to `main()`, just like in the original script. It must be possible to (re)build without cloning or creating a symbolic link or any other once-off setup step.

Just copy the logic of the original script for now.

Reply:
Logical changes already started, you know if I already spend time resolving this matters why should I drop it? Its my work after all. If you wish to rebuild simply do
`./scripts/xtensa-build-zephyr.py -p <your_SOF_dir>/zephyrproject -v -v <platform>`
Maybe I will add new logic for convenience - if -p flag is empty then resolve it to <your_SOF_dir>/zephyrproject by default. The only use case with providing paths I can think of is when you would clone multiple zephyr repositories under different names into the SOF_TOP directory and would like to point to different ones lets say building platformX with zephyrproject1_dir and building platformY with zephyrproject2_dir. I assume it was original goal of the xtensa-build-zephyr.sh and leave it like this.

Reply:
I really don't see which actual problems are solved by moving the single, `build_platforms()` line out of `main()`. The only effect I've seen so far has been to cost me at least 15 min trying to understand why nothing builds. Not something we want every user to experience and it would cost a lot more in validation.